lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3503,10 +3503,12 @@ static bool typeCheckDerivativeAttr(ASTContext &Ctx, Decl *D,
       };
 
   auto isValidOriginal = [&](AbstractFunctionDecl *originalCandidate) {
-    return checkFunctionSignature(
-        cast<AnyFunctionType>(originalFnType->getCanonicalType()),
-        originalCandidate->getInterfaceType()->getCanonicalType(),
-        checkGenericSignatureSatisfied);
+    // TODO(TF-982): Allow derivatives on protocol requirements.
+    return !isa<ProtocolDecl>(originalCandidate->getDeclContext()) &&
```

Comment:
Minor: I'd prefer to split the "protocol requirement" condition into an `if` statement, to simplify the condition:
```cpp
// TODO(TF-982): Allow derivatives on protocol requirements.
if (isa<ProtocolDecl>(originalCandidate->getDeclContext())
  return false;
return checkFunctionSignature(
  cast<AnyFunctionType>(originalFnType->getCanonicalType()),
  originalCandidate->getInterfaceType()->getCanonicalType(),
  checkGenericSignatureSatisfied);
```